### PR TITLE
Feature/gym 25 create security config

### PR DESCRIPTION
**Pull request by Ka1zerov**
added security config
added registration logic
added JWT support
added CurrentUserService (available for any module)
!!! IMPORTANT -> I did not add oauth2 clients yet (needs to be done in order for only our frontend to have access to our backend)

### CODE REVIEW

**Review by Valentine-456 — COMMENTED** [review]
Generally looks good, but we need to discuss how does it work on a daily meeting

File: src/main/java/com/sourcery/gymapp/backend/authentication/jwt/JwtConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        // Create a JWKSource from the JWK
+        JWKSource<SecurityContext> jwkSource = new ImmutableJWKSet<>(new JWKSet(jwk));
+        return new NimbusJwtEncoder(jwkSource);
+    }
+
+    @Bean
+    public JwtDecoder jwtDecoder() {
+        // Configure the JWT decoder with the public key to verify the token's signature
+        return NimbusJwtDecoder.withPublicKey(publicKey).build();
```
Comment (Valentine-456) [inline src/main/java/com/sourcery/gymapp/backend/authentication/jwt/JwtConfig.java:45]: TODO: remove comments from the code

Comment (Ka1zerov) [reply src/main/java/com/sourcery/gymapp/backend/authentication/jwt/JwtConfig.java:42]: done

**Review by Valentine-456 — APPROVED** [review]
LGTM, I approve, but it needs a discussion with a whole team about how it works on a daily.
I think we can connect it to unblock all other tasks

**Pull request merged**